go/plugins/googlecloudsql/postgres/genkit.go

```diff
@@ -0,0 +1,120 @@
+package postgres
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/firebase/genkit/go/ai"
+	"github.com/firebase/genkit/go/genkit"
+)
+
+// The provider used in the registry.
+const provider = "googlecloudsql-postgres"
+
+// GoogleCloudSQLPostgres holds the current plugin state.
+type GoogleCloudSQLPostgres struct {
+	Config      EngineConfig
```

Comment:
We will need a RetrieverConfig instead of the EngineConfig with variables: engine, table_name, schema name, content column, embedding column, metadata columns, ignore metadata columns, id column, metadata json column, distance strategy, and k